### ISSUE TITLE
Make the AI take invasion decisions when appropriate

### ISSUE THREAD

**Issue by BriceLeduc** [MattTheLegoman/RealmsInExile#642]
At the moment the AI seems to take invasion decisions (Destroy Gondor/Mordor and Castamiri claim) whenever they fill the requirements.
While this is rarely an issue for "Destroy" because by that point either actor is strong enough to win, the Castamiri claim is often pressed while Anadune has no realistic hope of winning.

**Comment by JukeJukestar** [comment]
I agree completely. Sauron should carefully plan his moves before swooping in for the final assault.

Castamiri I would be a bit more generous though, they still shouldnt declare when they are hopelessly outmatched but I can see them trying even if Gondor is still stronger than them, but in a different war or something.

**Comment by RedArkady** [comment]
A related issue to this is that Gondor doesn't seem to be motivated by some decisions available to it. Even when it has overwhelming military forces it doesn't seem interested in pursuing Integrate Andrast, Harondor, or Dor Rhunen - if it gets close to achieving these it's only due to aggressive vassals. This may be linked to the issue of Aragorn being _hopelessly_ cautious for some reason.

**Comment by jj248** [comment]
I've now made it so that the AI _should_ only declare war if its military strength outweighs that of that of the target (accounting for allies and non-permanent tributaries on both sides!).

Time to test to see if it's worked!

**Comment by savagesach** [comment]
I've found over several observer games that mordor no longer uses the destroy gondor decision against gondor, instead using kingdom holy wars. I'd suggest that at least mordor should take their decision whenever available.

**Comment by Jauffr** [comment]
Found the issue here. The AI does take the decision, however Sauron won't call his tributaries in. As a result he always loses this war and only takes over gondor by doing standard holy wars
![Screenshot 2023-03-22 180446](https://user-images.githubusercontent.com/108106660/227065718-c6f464d9-9fa4-488d-8a0f-85a263c3be37.png)
![Screenshot 2023-03-22 180122](https://user-images.githubusercontent.com/108106660/227065722-45977203-4a63-44db-a293-d7dc2b9084f0.png)

**Comment by savagesach** [comment]
I've run several observer games and I can confirm ai aragorn uses destroy mordor when he's not supposed to be able to (yes I know this is a temporary fix) and ai sauron very rarely uses the destroy gondor decision, instead using kingdom holy wars.
